core/src/main/java/org/apache/calcite/schema/Schemas.java

```diff
@@ -224,15 +224,15 @@ public static DataContext createDataContext(
    * array. */
   public static Enumerable<Object[]> enumerable(final FilterableTable table,
       final DataContext root) {
-    return table.scan(root, ImmutableList.of());
+    return table.scan(root, new ArrayList<>());
   }
 
   /** Returns an {@link org.apache.calcite.linq4j.Enumerable} over the rows of
    * a given table, not applying any filters and projecting all columns,
    * representing each row as an object array. */
   public static Enumerable<Object[]> enumerable(
       final ProjectableFilterableTable table, final DataContext root) {
-    return table.scan(root, ImmutableList.of(),
+    return table.scan(root, new ArrayList<>(),
         identity(table.getRowType(root.getTypeFactory()).getFieldCount()));
```

Comment:
Why this change ?

Reply:
I have the same question. If `ImmutableList.of()` is bad, then we should forbid its use, and move to `Collections.emptyList()`.

Reply:
See the commit comment. That parameter to `scan` is mutable, always has been. The implementation calls `List.removeIf` on it. As of guava-28.2-jre, `removeIf` throws even if the list is empty. So, these calls were incorrect to pass an immutable list.

Reply:
@julianhyde , `Collection.emptyList()` supports `removeIf` just fine.

Reply:
But it's not mutable. The contract calls to a mutable list.

Reply:
If you think the contract is to treat the list as fully mutable, then, adjust the Javadoc, and adjust the implementation so it handles the mutations of the list.

In other words, for now you pass an ArrayList and expect it to be mutated. However, you seem to ignore the mutated value of the ArrayList, so the new code might result in invalid data being returned.